### PR TITLE
Set closed_date to null when a trial is re/opened

### PR DESCRIPTION
**Pull request by Vegemash**

### CODE REVIEW

File: models/Trial.php [hunk 1]
```diff
@@ -615,7 +615,10 @@ public function removeUserPermission($permission_id)
      */
     public function transitionState($new_state)
     {
-        if ((int)$this->status === Trial::STATUS_OPEN && (int)$new_state === Trial::STATUS_IN_PROGRESS && $this->hasShortlistedPatients()) {
+        $trial_status = (int)$this->status;
+        $new_state = (int)$new_state;
```
Comment (apigram) [inline models/Trial.php]: Don't think it's good practice to have a local variable with the exact same name as a parameter. Parameters should be read-only unless they are references!

Comment (Vegemash) [reply models/Trial.php]: I'm not adding a new variable with the same name. I'm effectively casting it to a type so we don't have to do it over and over again in an ugly fashion in the rest of the function.

Comment (apigram) [reply models/Trial.php]: Still not good practice to mutate a parameter. They are meant to be read-only!

Comment (apigram) [reply models/Trial.php]: If you want to avoid this, you could probably just change the PHPDoc to indicate its an int.

**Pull request merged**